Rename internal functions.

bangerth opened the pull request:
https://github.com/geodynamics/aspect/pull/5474 illustrates that an inadvertant addition of `virtual` to an internal function leads to a complete disaster preventing anyone from deriving from that class in a meaningful way. Here's the summary:

The base class, at least before https://github.com/dealii/dealii/pull/16242 basically looked like this:
```
class X {   // really SphericalManifold
  public:
    virtual void f();
  private:
    virtual void f(int);
};
```
If you derive from it, you don't care about the `private` member function of the same name as the `public` one. It's none of our business. But if we overload only the `public` one, like in
```
class Y : public X {
  public:
    virtual void f() override;
};
```
then the compiler says that this *hides* the other member function, and the warning is converted to an error. So one tries this:
```
class Y : public X {
  public:
    virtual void f() override;
  private:
    virtual void f(int) override;
};
```
This silences the warning about hiding the other function, but now we don't know how to *implement* the overload: It should really just call the function in the base class, but that function is `private` and so we can't do
```
void Y::f(int i) { return X::f(i); }
```

It just isn't possible to derive correctly from a base class that has a `virtual` `private` member function of the same name `virtual` `public` member function. We just shot ourselves in the foot by declaring that internal function as `virtual` by accident :-(

#16242 already removes the `virtual` from the offending function declaration, but I continue to find it awkward that we now have `virtual` and non-`virtual` functions of the same name. Since name lookup happens without regard to access specifiers (`public`, `protected`, `private`), *and only then does a compiler determine whether the function is accessible*, this is not a great design. The patch therefore renames the internal functions so that they have different names than the `public` `virtual` ones.

drwells commented on source/grid/manifold_lib.cc:
```diff
@@ -1000,7 +1000,7 @@ SphericalManifold<dim, spacedim>::get_new_point(
 template <>
 Point<3>
-SphericalManifold<1, 3>::get_new_point(
+SphericalManifold<1, 3>::my_get_new_point(
```
I'm fine with the rename, but - why do these functions even exist? All of them (except the template base-case) just call `do_get_new_point()`. Could you check if they can be deleted (perhaps after adding an `if constexpr` check)?

bangerth commented:
OK, this seems to work -- the two CUDA checks are due to timeouts in downloading some package.

I renamed one of the `my_*` functions to `do_*` and removed the other one. Nice catch, @masterleinad. The existing `do_get_new_point()` function was in the global namespace (actually, an anonymous namespace), so I moved that into an anonymous namespace `internal::SphericalManifold::{}` because the name of the function does not relate to the class it supports, and there are other classes in this file as well.

I think this is ready from my side.

drwells reviewed (approved):
Much nicer!
